modules/functional_expansion_tools/include/utils/Hashing.h

```diff
@@ -0,0 +1,93 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef HASHING_H
+#define HASHING_H
+
+// MOOSE includes
+#include "MooseTypes.h"
+
+/**
+ * This namespace provides efficient algorithms for quickly hashing different types for checking
+ * identity with a very low collision probability.
+ */
+namespace hashing
+{
+typedef std::size_t HashValue;
+
+/**
+ * Final iteration of the variadic template with no additional arguments
+ */
+inline void
+hashCombine(HashValue & /* seed */)
+{
+}
+
+/**
+ * Variadic template to hashing a combination with finite size
+ *   see: https://stackoverflow.com/a/38140932
+ */
+template <class T, class... Rest>
+inline void
+hashCombine(HashValue & seed, const T & value, Rest... rest)
+{
+  std::hash<T> hasher;
+
+  seed ^= hasher(value) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+  hashCombine(seed, rest...);
+}
+
+/**
+ * Hash function for sampling 10 points from a large container and hashing
+ *   see: https://stackoverflow.com/a/37007715
+ */
+template <class Container>
+HashValue
+hashLargeContainer(Container const & container)
+{
+  std::size_t size = container.size();
+  std::size_t stride = 1 + size / 10;
+  HashValue value = size;
+
+  for (std::size_t i = 0; i < size; i += stride)
+  {
+    hashCombine(value, container.data()[i]);
+  }
+
+  return value;
+}
+
+/**
+ * Hashing for Point
+ */
+inline HashValue
+hashCombine(const libMesh::Point & point)
+{
+  HashValue value = 3;
```

Comment:
Does something need changed?

Reply:
A comment that explains why 3. Why not 33? Or 333? :-)

Reply:
Or 42? :-)

I put in a comment. It's just a seed value, and provides entropy against the other seed value.

Reply:
Also, this would be much better:
```
HashValue seed = 3;
```
the word `value` is just too general. Moreover, notice that here you use 3 and pass 3 parameters and below, you use 4 and pass 4 parameters.

Reply:
Got it `seed` it is.

I didn't even notice the coincidence with the number of arguments.